src/control/server/mgmt_pool.go

```diff
@@ -653,6 +655,9 @@ func (svc *mgmtSvc) PoolGetACL(ctx context.Context, req *mgmtpb.GetACLReq) (*mgm
 
 // PoolOverwriteACL forwards a request to the IO server to overwrite a pool's Access Control List
 func (svc *mgmtSvc) PoolOverwriteACL(ctx context.Context, req *mgmtpb.ModifyACLReq) (*mgmtpb.ACLResp, error) {
+	if err := svc.checkReplicaRequest(req); err != nil {
```

Comment:
I'm a little curious what the logic is for when we should be the leader vs. any replica-- looks like we should be the leader in any case when we're making changes to a pool. Is that correct? If so it seems like we should be the leader to modify the ACL.

Reply:
The distinction is a little mushy. This kind of operation doesn't need to run on the current MS leader because the ACL isn't persisted in the MS db, it's stored as a property in the pool rdb. So we only require that the RPC was directed at a MS replica so that it can do the PoolID -> svc_ranks lookup and then send the request down to the ioserver to do the cart RPC to the pool service leader. Doesn't matter where that cart RPC comes from.

In the past, we always did this sort of thing on the MS leader because there was only one MS replica. In a subsequent PR, I'm going to be pushing some updates to the MS client code to make it smarter about choosing a server for the request. The idea being that we'll randomly choose a server to try the request, and that server can either respond that it's not a replica (in which case it will send back the list of replicas), service the request if it's allowed to, or respond that it's not the leader and send back its understanding of the current leader, at which point the client should finally succeed. 

This scheme does potentially add some latency, but it makes configuration simpler and gives us more opportunity to do load balancing so that the MS leader isn't a bottleneck unless it absolutely needs to be.

Reply:
Cool. That makes sense to me. Thanks!